### PR TITLE
GEN-53: Upgrade `error-stack` toolchain and use `Error::provide`

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

The Error::provide API has changed. To avoid conflicts with the recent Rust toolchain and maintain compatibility we need to update the toolchain. This reverts #2906 (GEN-52).

### 🚢 Has this modified a publishable library?

This PR:

- [x] modifies a **Cargo**-publishable library, but **it is not yet ready to publish**

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] require changes to docs which **are made** as part of this PR

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

### CODE REVIEW

**Review by indietyp — APPROVED** [review]
Even tho our bot overlords have blessed this PR here you go. Overall I dislike that the provide API has been removed, but o well 🤷 it's a nightly API after all.

File: .github/renovate.json [hunk 1]
```diff
@@ -205,6 +205,11 @@
       "matchPackagePatterns": ["^tracing[-_]?"],
       "excludePackageNames": ["tracing-opentelemetry"],
       "groupName": "`tracing` crates"
+    },
```
Comment (indietyp) [inline .github/renovate.json:208]: Specific reason why anyhow is being singled out into it's own group?

Comment (indietyp) [reply .github/renovate.json:208]: (Non blocking just curious)

Comment (TimDiekmann) [reply .github/renovate.json:208]: Renovate tries updating `anyhow` all the time but we only have a patch version specified here for nightly compatibility. After all this is a library and `anyhow` is an optional feature and we should not put stricter requirements on it than needed.

**Review by indietyp — APPROVED** [review]

**Pull request merged**